mm2src/lp_ordermatch.rs

```diff
@@ -725,6 +773,30 @@ fn test_parse_orderbook_pair_from_topic() {
 
 async fn maker_order_created_p2p_notify(ctx: MmArc, order: &MakerOrder) {
     let topic = orderbook_topic_from_base_rel(&order.base, &order.rel);
+    let base_coin = match lp_coinfind(&ctx, &order.base).await {
```

Comment:
Please add `base_protocol_info` and `rel_protocol_info` args instead - in most use cases the coins are already found before `maker_order_created_p2p_notify` is called. Please use a compact `lp_coins::find_pair` whenever it will be required to search for them.

Reply:
Do you mean adding `base_protocol_info` and `rel_protocol_info` args to MakerOrder Struct?

Reply:
No, I mean changing the signature of `maker_order_created_p2p_notify` to
```rust
async fn maker_order_created_p2p_notify(ctx: MmArc, order: &MakerOrder, base_protocol_info: Option<Vec<u8>>, rel_protocol_info: Option<Vec<u8>>)
```

As I can see, in 2 of 3 use cases coins are already found before this one is called. In the 3rd case, you can use `find_pair`.

I would have said *fields* if I meant adding them to the struct :slightly_smiling_face: It could have also caused problems because order can stay alive between MM2 restarts. If the coin is reactivated with a different address format then `MakerOrder::base_protocol_info` would contain invalid info.